### DISCUSSION TITLE
有没有考虑过把MySQL迁移到PostgreSQL

### DISCUSSION THREAD

**Discussion by Aloento** [bangumi/server#191] [Ideas]
如题，自己使用PG下来感觉是真的爽，也在很多项目中见到过大家都逐渐选了PG

**Comment by trim21** [comment]
比较难办，得先把旧功能都迁移过来，放弃掉旧代码库再考虑

**Reply by Aloento** [reply]
大佬不睡觉的吗）
有机会的话可以叫我，我有迁移经验

**Reply by trim21** [reply]
主要是现阶段得兼容旧代码库，旧代码库这边只有sai老板在写代码，无论是什么改动都很麻烦。

**Reply by Aloento** [reply]
咱们新站目前也没有上线（  
所以不妨可以考虑一个breaking change  
还是说我们以后需要新老站点并行？

**Reply by trim21** [reply]
起码得实现了旧站的功能再把旧站下线吧。下线之前都得兼容。按照这个开发速度，可能是只有break没有change。

至于是一次性更新还是按照页面分别更新还没想好。

**Reply by trim21** [reply]
现在两边session也不通用，部分替换升级的话用户体验也比较差（）

**Reply by Aloento** [reply]
如果你要一次性更新那就可以change，如果分别更新那这确实就不太可能change了，但是这样真的好吗（

**Reply by Aloento** [reply]
我之前为客户重构项目的时候采取的方案是整套替换，感觉这样实现会自由许多，不需要考虑曾经的一堆包袱
_（因为以前的项目有时候的设计真的很shit_

**Reply by trim21** [reply]
现在后端大概才做完了1/4，前端只做了登录。

要替换还早着呢（）

**Reply by Aloento** [reply]
咱们有任务的时间规划吗，还是说做多久算多久（

**Reply by trim21** [reply]
那大家都不领工资，当然没有规划啦 b38

**Reply by Aloento** [reply]
那不就得了，这么说下来答案不就很明显了，只能一次性替换了
breaking change，请 _（迫切_

**Reply by trim21** [reply]
我其实没明白你是想break什么…

**Reply by Aloento** [reply]
各种break吧，不过这个thread讨论的是break数据库
可以不用太考虑老数据库，直接按新的pg来写，最后迁移的时候写数据库迁移代码就好
你觉得这个有可行性吗

**Reply by trim21** [reply]
我只是介绍一下现在的情况，不代表支持或者反对迁移（）

现在兼容还有希望平行上线，要是等开发完迁移可能就遥遥无期了。

一个是开发本身也慢（到现在半年了差不多才写了1/4），一个是sai老板也很忙。

比如现在数据库是sai老板在管理，之前说搞个主从就已经半个月过去了…

然后再前一个问题是一个月还是两个月前说要给时间线还是哪个东西加软删除，然后sai老板一直在忙。

总之大家都是用爱发电，旧站又不是不能用 b38

**Reply by Aloento** [reply]
好家伙，那这个问题得搁置了
真实 又不是不能用（

**Reply by Aloento** [reply]
trim老板有什么我可以干的吗最近比较闲，我看了一圈issue和project下来发现好像列出来的内容都有pr了

**Reply by trim21** [reply]
比如日志功能？ https://github.com/bangumi/dev-docs/issues/1

**Reply by Aloento** [reply]
让我康康，翻了半天确实没往docs里面翻****

**Reply by trim21** [reply]
我也的确没写在readme里，刚刚才加上 b38

**Reply by trim21** [reply]
https://bgm.tv/group/topic/371666

这个是你吗 （）

**Reply by Aloento** [reply]
啊？这个不是我，我还没工作呢
不过我也是主要写C#的，如果他真写的话我可能会考虑加入
一周追上目前进度是不是太快了点（）我觉得怎么也得半个月吧

**Reply by Aloento** [reply]
说实话他这么一说我就很感兴趣了，老板你怎么看

**Reply by trim21** [reply]
我不会C#

**Reply by Aloento** [reply]
我看你会TS，C#和TS一样的
语法一致，就是C#多了一些功能，比如自带的数据库ORM查询LINQ之类的
你现在用的GORM Gen的语法和LINQ差不多

**Reply by Aloento** [reply]
我很激动，我想试试看（（（
